tests/functional/profiles/test_profile_dir.py

```diff
@@ -106,8 +100,8 @@ def dbt_debug(self, project_dir_cli_arg=None, profiles_dir_cli_arg=None):
         if profiles_dir_cli_arg:
             command.extend(["--profiles-dir", str(profiles_dir_cli_arg)])
 
-        # get the output of `dbt debug` regarless of the exit code
-        return run_dbt_and_capture_stdout(command, expect_pass=None)
+        # get the output of `dbt debug` regardless of the exit code
+        return run_dbt_and_capture(command, expect_pass=False, capture_stderr=True)
```

Comment:
I'm the one who originally wrote this section of code.

Good catch on my misspelling of "regardless"! 🙏 

I didn't confirm this one way or the other, but my naive expectation is that we would preserve `expect_pass=None`. This is because this test intends to inspect the output of the `dbt debug` command _regardless_ if the exit code is zero or non-zero (corresponding to PASS vs FAIL, respectively).

Put another way, `dbt debug` will pass in some test scenarios and fail in others, but in both cases, we don't care about the pass/fail status; we only care about the text in stdout. If we uniformly expect_pass=False or expect_pass=True, it wouldn't allow us to reach all the test cases in the desired manner.

```suggestion
        # get the output of `dbt debug` regardless of the exit code
        return run_dbt_and_capture(command, expect_pass=None, capture_stderr=True)
```

It might also be that the refactor of `print` statements in `dbt debug` allows and/or requires us to refactor this test as well. If so, we'll just want to carefully consider things like `expect_pass=False`.

Reply:
This is interesting. I'll have to double check this. My thinking was "None" and "False" are evaluated the same way in Python `if`-`else` blocks. I'll dive a bit deeper into the code here to ensure there isn't something else going on behind the scenes.

Reply:
Oh I didn't know we had this feature. This is great and there were several times I wish I had known about this sooner! 
```python
if expect_pass is not None:
    assert success == expect_pass, "dbt exit state did not match expected"
```
Thanks for correcting me. I'll fix this back.